### PR TITLE
Check that input is array_like, if required

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Related to #79.

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@MakisH @IshaanDesai Note: I will implement `check_array_like` for all functions that actually accept a `array_like`, if you agree with the general approach. Just don't want to roll this out to the whole bindings just to revert/change it. Let's prototype this first at the example of `read_block_scalar_data` and `write_block_scalar_data`

**Comment by BenjaminRodenberg** [comment]
You can see the actual error here: https://github.com/precice/python-bindings/pull/80/checks?check_run_id=1710161099#step:7:122

**Review by MakisH — APPROVED** [review]
This error message is very helpful and much better than before! :+1: 

I have not tried running the code. The target branch looks correct. Maybe squash your two commits.

**Comment by BenjaminRodenberg** [comment]
Now all API functions are fixed in the same fashion. Waiting for the review of @IshaanDesai. I will squash everything into one single commit :+1:

File: cyprecice/cyprecice.pyx [hunk 1]
```diff
@@ -568,6 +582,11 @@ cdef class Interface:
         >>> positions.shape
         (5, 3)
         """
+        check_array_like(vertex_ids, "vertex_ids", "get_mesh_vertices")
+
+        if not isinstance(vertex_ids, np.ndarray):
```
Comment (IshaanDesai) [inline cyprecice/cyprecice.pyx:587]:
```suggestion
        if not isinstance(positions, np.ndarray):
```

File: cyprecice/cyprecice.pyx [hunk 2]
```diff
@@ -568,6 +582,11 @@ cdef class Interface:
...
+            positions = np.asarray(vertex_ids)
```
Comment (IshaanDesai) [inline cyprecice/cyprecice.pyx:588]: Is this necessary?

**Review by IshaanDesai — APPROVED** [review]

File: cyprecice/cyprecice.pyx (hunk 2 above)
Comment (BenjaminRodenberg) [reply cyprecice/cyprecice.pyx:588]: Good eye! It's not. `np.ascontiguousarray` already takes care of converting everything that we get into a `np.ndarray`.

Comment (BenjaminRodenberg) [reply cyprecice/cyprecice.pyx:588]: Removed this line and tests still work. The test `test_set_mesh_vertex_list` should be testing exactly this case.

Comment (BenjaminRodenberg) [reply cyprecice/cyprecice.pyx]: Took care of this case an similar ones in c065016d0fe622e8b5c4de9754289316c7c046dd. 

Some insights: Converting to a numpy array is only neaded, of `input.shape` is required before converting `input` via `np.ascontiguousarray`. This is only the case for vector valued input (vector data and positions), where we want to make sure that number of dimensions and samples fit with the corresponding ids.

**Pull request merged**